src/Service/EolValidator.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Service;
+
+use Carbon\Carbon;
+use Composer\Semver\Semver;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Magento\MagentoCloud\Filesystem\FileList;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Service EOL validator.
+ *
+ * Class EolValidator
+ */
+class EolValidator
+{
+    /**
+     * Set the notification period.
+     */
+    private const NOTIFICATION_PERIOD = 3;
+
+    /**
+     * @var FileList
+     */
+    private $fileList;
+
+    /**
+     * @var ServiceFactory
+     */
+    private $serviceFactory;
+
+    /**
+     * @var ElasticSearch
+     */
+    private $elasticSearch;
+
+    /**
+     * @var array
+     */
+    private $services = [
+        ServiceInterface::NAME_PHP,
+        ServiceInterface::NAME_ELASTICSEARCH,
+        ServiceInterface::NAME_RABBITMQ,
+        ServiceInterface::NAME_REDIS,
+        ServiceInterface::NAME_DB
+    ];
+
+    /**
+     * EolValidator constructor.
+     *
+     * @param FileList $fileList
+     * @param ServiceFactory $serviceFactory
+     * @param ElasticSearch $elasticSearch
+     */
+    public function __construct(
+        FileList $fileList,
+        ServiceFactory $serviceFactory,
+        ElasticSearch $elasticSearch
+    ) {
+        $this->fileList = $fileList;
+        $this->serviceFactory = $serviceFactory;
+        $this->elasticSearch = $elasticSearch;
+    }
+
+    /**
+     * Validate the EOL of a given service and version by error level.
+     *
+     * @param int $errorLevel
+     * @return array
+     * @throws ServiceMismatchException
+     */
+    public function validateServiceEol(int $errorLevel): array
```

Comment:
I propose to not pass `$errorLevel` in this method, as it adds more complexity.
You can return array of all errors with error level as a key:
```
$errors = [
      'level_warning` => [
        errors...
      ],
       'level_notice` => [
           errors...
      ],
]
```